manifests/charon/plugin.pp

```diff
@@ -0,0 +1,16 @@
+# Configure charon pluging
+define strongswan::charon::plugin (
+  $options    = {},
```

Comment:
could you enforce the datatypes within the hash? I assume it's strings?
```suggestion
  Hash[String[1], String[1]] $options    = {},
```

Reply:
Not really, this can be a nested data structure and and up with the configuration like:
```
name {
        section-one {
                 somevalue = asdf
                 subsection {
                      othervalue = xxx
                 }
                 yetanother = zz
        }
        section-two {
                 x = 12
            }
}
```

Reply:
Okay, but we can at least enforce Hash?

```suggestion
  Hash $options    = {},
```

Reply:
Even though the template will reject anything else, `Hash` would still be picked up by puppet-strings (in due course), so is worth it.